### PR TITLE
Run selected tests in en-US culture

### PR DESCRIPTION
**Pull request by RussKie**
Mu OS default locale is set to "en-AU", and these tests kept failing for me. Add two new XUnit attributes to specify the desired locale.

/cc: @weltkante @vladimir-krestov I know that in the past you had troubles with some locale-dependent tests on your ends. If you could run these changes and confirm that (at least) these tests pass for you - it would be awesome. 
Also, if you see any other tests that may benefit from the new decoration - please let me know (or sent a PR in :)).
Thank you.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6985)

### CODE REVIEW

**Comment by weltkante** [comment]
Thanks, I'll cross reference with my local branch I pull in to adjust broken tests (an older version is also [here](https://github.com/dotnet/winforms/compare/main...weltkante:culture) but I haven't pushed it for a long time, that was just for commenting on an issue back then). Do you want this to be the resolution for #2734? If so I can do a PR to adress the affected tests.

File: src/Common/tests/TestUtilities/XUnit/UseCultureAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        /// and replaces them with the new cultures defined in the constructor.
+        /// </summary>
+        /// <param name="methodUnderTest">The method under test</param>
+        public override unsafe void Before(MethodInfo methodUnderTest)
+        {
+            _originalCulture = Thread.CurrentThread.CurrentCulture;
+            _originalUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            CultureInfo.DefaultThreadCurrentCulture =
+                Thread.CurrentThread.CurrentCulture = Culture;
```
Comment (Tanya-Solyanik) [inline src/Common/tests/TestUtilities/XUnit/UseCultureAttribute.cs:74]:
```suggestion
            CultureInfo.DefaultThreadCurrentCulture = Culture;
            Thread.CurrentThread.CurrentCulture = Culture;
```

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good.

File: src/Common/tests/TestUtilities/XUnit/UseCultureAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            Thread.CurrentThread.CurrentUICulture = UICulture;
+
+            if (SetUnmanagedUiThreadCulture)
```
Comment (dreddy-work) [inline src/Common/tests/TestUtilities/XUnit/UseCultureAttribute.cs:77]: you may want to check current culture before setting it, if it needs to be set.

**Pull request merged**